core/cloudflow-operator/src/main/scala/cloudflow/operator/event/StreamletChangeEvent.scala

```diff
@@ -172,4 +185,97 @@ object StreamletChangeEvent {
         case _ ⇒ Nil // app could not be found, do nothing.
       }
       .mapConcat(_.toList)
+
+  def toInputConfigUpdateAction(
+      implicit system: ActorSystem,
+      ctx: DeploymentContext
+  ): Flow[(Option[CloudflowApplication.CR], StreamletChangeEvent[Secret]), Action[ObjectResource], NotUsed] =
+    Flow[(Option[CloudflowApplication.CR], StreamletChangeEvent[Secret])]
+      .map {
+        case (Some(app), streamletChangeEvent) ⇒
+          import streamletChangeEvent._
+          app.spec.deployments
+            .find(_.streamletName == streamletName)
+            .map { streamletDeployment ⇒
+              val secret: Secret = streamletChangeEvent.watchEvent._object
+              system.log.info(s"Detected input secret '${secret.metadata.name}' changed for streamlet '$streamletName'.")
+
+              secret.data
+                .get(ConfigInputChangeEvent.SecretDataKey)
+                .map { bytes =>
+                  val config = ConfigFactory.parseString(new String(bytes, StandardCharsets.UTF_8))
+
+                  // TODO refactor
```

Comment:
This will get neater once the kubernetes, flink, spark settings are added, good for now.